scintillometry/io/psrfits/hdu.py

```diff
@@ -97,18 +123,43 @@ def frequency(self):
                          ((n_chan + 1) // 2)) * chan_bw
         return u.Quantity(freq, u.MHz, copy=False)
 
+    @frequency.setter
+    def frequency(self, freq):
+        """Frequency setter. The input should be the frequency array."""
+        freq = freq.to(u.MHz)
+        n_chan = len(freq)
+        # TODO how do we deal with the even number of nchan
```

Comment:
As I understand the PSRFITS docs (see comment above), we assume channel 0 is missing (and we have to enforce this from wherever we call the frequency setter). So, I think I'd mirror the getter, by doing `c_chan = freq[(n_chan + 1) // 2]` (This certainly is how I think you should test it: set with frequencies you get from a known header, and check that 'OBSFREQ' remains the same.